integrations/ConvertFlow/utils.js

```diff
@@ -0,0 +1,33 @@
+// default mapping for the events
+const standardEventsListMapping = {
+  cfReady: "CTA Ready",
+  cfView: "CTA Viewed",
+  cfConversion: "CTA Converted",
+  cfCompletion: "CTA Completed",
+  cfSubmit: "CTA Form Submitted",
+  cfAddToCart: "Product Added to Cart",
+  cfClosed: "CTA Closed",
+};
+
+/**
+ * This function is used to make a track Call to rudderstack.
+ * @param {*} standardEventsMap - mapping of events done by the user
+ * @param {*} eventName - standard event name
+ * @param {*} data
+ */
+const makeACall = (standardEventsMap, eventName, data) => {
+  const eventNames = Object.keys(standardEventsMap);
+  let updatedEvent = standardEventsListMapping[eventName];
+  eventNames.forEach((event) => {
+    if (standardEventsMap[event].includes(eventName)) {
+      updatedEvent = event;
+    }
+  });
+  window.rudderanalytics.track(updatedEvent, {
+    name: data.cta.name,
+    cta_type: data.cta.cta_type,
+    id: data.cta.id,
+  });
```

Comment:
We should put a check if data.cta is there or not.

Reply:
Can we use optional chaining?

Reply:
Yeah but in that case, undefined will be sent. We should check if the data is defined or not. It may be possible for some callbacks they are not sending data.

Reply:
updated